ktor-network/jvm/test/io/ktor/network/sockets/tests/ClientSocketTest.kt

```diff
@@ -91,6 +92,53 @@ class ClientSocketTest {
         }
     }
 
+    @Test
+    fun testSelfConnect() {
+        runBlocking {
+            // Find a port that would be used as a local address.
+            val port = getAvailablePort()
+
+            val tcpSocketBuilder = aSocket(ActorSelectorManager(Dispatchers.IO)).tcp()
+            // Try to connect to that address repeatedly.
+            for (i in 0 until 100000) {
+                try {
+                    val socket = tcpSocketBuilder.connect(InetSocketAddress("127.0.0.1", port))
+                    assertTrue(false, "connect to self succeed: ${socket.localAddress} to ${socket.remoteAddress}")
```

Comment:
fail("connect to self succeed: ${socket.localAddress} to ${socket.remoteAddress}")

Reply:
ok, i will fix this